PokemonTabletopAdventures_v3/PTA3.html

```diff
@@ -2757,4 +2783,229 @@ <h2>Configuration Options</h2>
+    "diplomacy", "diplomacytalent1", "diplomacytalent2",
+    "engineering", "engineeringtalent1", "engineeringtalent2",
+    "history", "historytalent1", "historytalent2",
+    "insight", "insighttalent1", "insighttalent2",
+    "investigate", "investigatetalent1", "investigatetalent2",
+    "medicine", "medicinetalent1", "medicinetalent2",
+    "nature", "naturetalent1", "naturetalent2",
+    "perception", "perceptiontalent1", "perceptiontalent2",
+    "perform", "performtalent1", "performtalent2",
+    "handling", "handlingtalent1", "handlingtalent2",
+    "programming", "programmingtalent1", "programmingtalent2",
+    "sleightofhand", "sleightofhandtalent1", "sleightofhandtalent2",
+    "stealth", "stealthtalent1", "stealthtalent2",
+    // Move Bonuses
+    "move_bonusaccuracy", "move_bonusattack", "move_bonusspecialattack", "move_critthreshold",
+    // Text boxes
+    "pokemonskills", "statpassives", "otherpassives", "notes", "inventory", "honors", "ownedpokemon",
+    // Origin
+    "originfeaturename", "originfeaturedesc"
+  ];
+
+  const r20export_features = ["classfeaturename", "classfeaturedesc"];
+
+  const r20export_moves = [
+    "move_name", "move_range", "move_type", "move_category", "move_frequency", "move_dicenumber", "damage_dice",
+    "move_acmod", "move_damagebonus", "move_template", "move_effect1_threshold", "move_effect1_name",
+    "move_effect2_threshold", "move_effect2_name", "move_effect"
+  ];
+
+  on("change:json_import", function() {
+    getAttrs(["json_import"], function (values) {
+      const json = tryParseJson(values.json_import);
+      if(json){
+        const newCharacterData = parseCharacterData(json);
+        setAttrs(newCharacterData);
+      } else {
+        setAttrs({ ["json_import_response"]: "Error - JSON string appears invalid" });
+      }
+    });
+  });
+
+  on("clicked:generate-export", function() {
+    getSectionIDs("repeating_moves", function (moveIDs) {
+      getSectionIDs("repeating_features", function (featureIDs) {
+        const idmoves = moveIDs.flatMap(id =>
+            r20export_moves.map(moveAttr => `repeating_moves_${id}_${moveAttr}`)
+        );
+        const idfeatures = featureIDs.flatMap(id =>
+            r20export_features.map(featureAttr => `repeating_features_${id}_${featureAttr}`)
+        );
+        getAttrs([...r20export_attrs, ...idmoves, ...idfeatures], function (values) {
+          let topLevelAttrs = includeOnlyKeysInArray(values, r20export_attrs);
+          let moveAttrs = includeOnlyKeysInArray(values, idmoves);
+          let featureAttrs = includeOnlyKeysInArray(values, idfeatures);
+          let chunkedMoves = partitionObject(moveAttrs, r20export_moves);
+          let chunkedFeatures = partitionObject(featureAttrs, r20export_features);
+          topLevelAttrs["exportmoves"] = chunkedMoves;
+          topLevelAttrs["exportfeatures"] = chunkedFeatures;
+          topLevelAttrs["Source"] = "roll20 [" + new Date().toUTCString() + "]";
```

Comment:
Could use template strings here.
```suggestion
          topLevelAttrs["Source"] = `roll20 [${new Date().toUTCString()}]`;
```